apps/image-editor/src/js/component/resize.js

```diff
@@ -0,0 +1,118 @@
+import Component from '@/interface/component';
+import { componentNames } from '@/consts';
+
+/**
+ * Resize components
+ * @param {Graphics} graphics - Graphics instance
+ * @extends {Component}
+ * @class Resize
+ * @ignore
+ */
+class Resize extends Component {
+  constructor(graphics) {
+    super(componentNames.RESIZE, graphics);
+
+    /**
+     * Current dimensions
+     * @type {Object}
+     * @private
+     */
+    this._dimensions = null;
+
+    /**
+     * Original dimensions
+     * @type {Object}
+     * @private
+     */
+    this._originalDimensions = null;
+
+    /**
+     * Lock aspect ratio state
+     * @type {boolean}
+     * @private
+     */
+    this._lockState = false;
+  }
+
+  /**
+   * Get current dimensions
+   * @returns {object}
+   */
+  getCurrentDimensions() {
+    if (!this._dimensions) {
+      const canvasImage = this.getCanvasImage();
+      this._dimensions = {
+        width: canvasImage.width,
+        height: canvasImage.height,
```

Comment:
I think it would be good to use shorthand.
Maybe...
```js
const { width, height } = this.getCanvasImage();
this._dimensions = { width, height };
```